ci/jenkins/test.sh

```diff
@@ -429,6 +444,131 @@ function deliver_antrea_windows {
+        # Pull necessary images in advance to avoid transient error
+        for image in "${common_images[@]}"; do
+            ssh -o StrictHostKeyChecking=no -n jenkins@${IP} "ctr -n=k8s.io images pull ${image}" || true
+        done
+        # Pull necessary images for Kubernetes conformance test
+        for i in "${!k8s_images[@]}"; do
+            ssh -o StrictHostKeyChecking=no -n jenkins@${IP} "ctr -n=k8s.io images pull ${k8s_images[i]} && ctr -n=k8s.io images tag ${k8s_images[i]} ${e2e_images[i]}" || true
+        done
+    done
+
+    echo "===== Build Antrea Windows on Windows Jumper Node ====="
+    echo "==== Reverting Windows VM ${WIN_JUMPER} ====="
+    govc snapshot.revert -vm ${WIN_JUMPER} win-initial
+    winVMIP=""
+    for i in `seq 10`; do
+        winVMIP=$(govc vm.ip -wait=2m -a ${WIN_JUMPER})
+        if [[ $winVMIP != "" ]]; then
+            echo "Windows VM ${WIN_JUMPER} powered on"
+            break
+        fi
+        echo "Windows VM ${WIN_JUMPER} failed to power on"
+        govc vm.power -on ${WIN_JUMPER} || true
+    done
+    if [[ $winVMIP == "" ]]; then
+        echo "Windows VM ${WIN_JUMPER} didn't power on after 10 tries, exiting"
+        exit 1
+    fi
+    winVMIP=${winVMIP#*,}
+    # Windows VM is reverted to an old snapshot so computer date needs updating.
+    for i in `seq 24`; do
+        sleep 5
+        ssh -o StrictHostKeyChecking=no -n Administrator@${winVMIP} "W32tm /resync /force" | grep successfully && break
+    done
+    rm -f antrea-windows.tar.gz
+    # Compress antrea repo and copy it to a Windows node
+    mkdir -p jenkins
+    tar --exclude='./jenkins' -czf jenkins/antrea_repo.tar.gz -C "$(pwd)" .
+    for i in `seq 2`; do
+        timeout 2m scp -o StrictHostKeyChecking=no -T jenkins/antrea_repo.tar.gz Administrator@${winVMIP}: && break
+    done
+    ssh -o StrictHostKeyChecking=no -n Administrator@${winVMIP} "docker pull ${DOCKER_REGISTRY}/antrea/golang:${GO_VERSION}-nanoserver && docker tag ${DOCKER_REGISTRY}/antrea/golang:${GO_VERSION}-nanoserver golang:${GO_VERSION}-nanoserver"
+    ssh -o StrictHostKeyChecking=no -n Administrator@${winVMIP} "rm -rf antrea && mkdir antrea && cd antrea && tar -xzf ../antrea_repo.tar.gz > /dev/null && NO_PULL=${NO_PULL}; DOCKER_NETWORK=host make build-windows && docker save -o antrea-windows.tar antrea/antrea-windows:latest && gzip -f antrea-windows.tar" || true
+    for i in `seq 2`; do
+        timeout 2m scp -o StrictHostKeyChecking=no -T Administrator@${winVMIP}:antrea/antrea-windows.tar.gz . && break
+    done
+
+    echo "===== Deliver Antrea Windows to Windows worker nodes and pull necessary images on Windows worker nodes ====="
+    sed -i 's/if (!(Test-Path $AntreaAgentConfigPath))/if ($true)/' hack/windows/Helper.psm1
+    kubectl get nodes -o wide --no-headers=true | awk -v role="$CONTROL_PLANE_NODE_ROLE" '$3 !~ role && $1 ~ /win/ {print $1}' | while read WORKER_NAME; do
+        WIN_NAME=${WORKER_NAME}
+        revert_snapshot_windows
```

Comment:
It's not common to use env var to pass parameter, should use function parameter directly:
`revert_snapshot_windows ${WORKER_NAME}`

Reply:
Updated, thanks.